code/bmpman/bmpman.cpp

```diff
@@ -1663,6 +1773,58 @@ void bm_lock_ani(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte
 	anim_free(the_anim);
 }
 
+
+void bm_lock_apng(int handle, int bitmapnum, bitmap_entry *be, bitmap *bmp, ubyte bpp, ubyte flags) {
+
+	int first_frame = be->info.ani.first_frame;
+	int nframes = bm_bitmaps[first_frame].info.ani.num_frames;
+
+	apng::apng_ani* the_apng;
```

Comment:
This is a good idea, in fact I'd like to go even further & remove the use of new entirely... except I can't figure out how to get around the try..catch block scoping issues (at least without adding a new class ctor and then having multiple almost identical default values being set for all the class variables).  Any suggestions that I might have missed?